pkg/services/dummy/dummy.go

```diff
@@ -0,0 +1,215 @@
+// Presentation is a VerifiablePresentation without valid cryptographic proofs
+// It is only usable in non-strict mode.
+type Presentation struct {
+	contract.VerifiableCredentialBase
+	Proof Proof
+}
+
+// Proof holds the Proof generated from the dummy Signer
+type Proof struct {
+	// Proof type, mandatory
+	Type string
+	// Contract as how it was presented to the user
+	Contract string
+	// Initials form the signing means
+	Initials string
+	// Lastname form the signing means
+	Lastname string
+	// Birthdate form the signing means
+	Birthdate string
+	// Email form the signing means
+	Email string
+}
+
+type sessionPointer struct {
+	sessionID string
+}
+
+func (s sessionPointer) SessionID() string {
+	return s.sessionID
+}
+
+func (s sessionPointer) Payload() []byte {
+	return []byte("dummy")
+}
+
+func (s sessionPointer) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		SessionID string `json:"sessionID"`
+	}{SessionID: s.sessionID})
+}
+
+type signingSessionResult struct {
+	ID      string
+	State   string
+	Request string
+}
+
+func (d signingSessionResult) Status() string {
+	return d.State
+}
+
+func (d signingSessionResult) VerifiablePresentation() (contract.VerifiablePresentation, error) {
+	// todo: the contract template should be used to select the dummy attributes to add
+	// reqContract := d.Request
+
+	if d.Status() != SessionCompleted {
+		return nil, nil
+	}
+
+	return Presentation{
+		VerifiableCredentialBase: contract.VerifiableCredentialBase{
+			Context: []string{contract.VerifiableCredentialContext},
+			Type:    []string{contract.VerifiablePresentationType, VerifiablePresentationType},
+		},
+		Proof: Proof{
+			Type:      NoSignatureType,
+			Initials:  "I",
+			Lastname:  "Tester",
+			Birthdate: "1980-01-01",
+			Email:     "tester@example.com",
+			Contract:  d.Request,
+		},
+	}, nil
+}
+
+var errNotEnabled = errors.New("not allowed in strict mode")
+
+// VerifyVP verifies a verifiablePresentation provided by a byte array.
+// It returns the contract.VerificationResult.
+func (d Dummy) VerifyVP(rawVerifiablePresentation []byte) (*contract.VerificationResult, error) {
+	if d.InStrictMode {
+		return nil, errNotEnabled
+	}
+
+	ndp := Presentation{}
+	if err := json.Unmarshal(rawVerifiablePresentation, &ndp); err != nil {
+		return nil, err
+	}
+
+	c, err := contract.ParseContractString(ndp.Proof.Contract, contract.StandardContractTemplates)
+	if err != nil {
+		return nil, err
+	}
+
+	return &contract.VerificationResult{
+		State:          contract.Valid,
+		ContractFormat: ContractFormat,
+		DisclosedAttributes: map[string]string{
+			"initials":  ndp.Proof.Initials,
+			"lastname":  ndp.Proof.Lastname,
+			"birthdate": ndp.Proof.Birthdate,
+			"email":     ndp.Proof.Initials,
+		},
+		ContractAttributes: c.Params,
+	}, nil
+}
+
+// SigningSessionStatus looks up the session by the provided sessionID param.
+// When the session exists it returns the current state and advances the state to the next one.
```

Comment:
I think this comment should be on the interface, and the implementation shouldn't need (unless the implementation is very complex and needs explanation) additional comments on it. It risks those 2 forking apart